src/nvim/ex_cmds2.c

```diff
@@ -744,40 +744,38 @@ void dbg_breakpoint(char_u *name, linenr_T lnum)
 
 static char_u   *profile_fname = NULL;
 
-/*
- * ":profile cmd args"
- */
+/// :profile cmd args
 void ex_profile(exarg_T *eap)
 {
   static proftime_T pause_time;
 
-  char_u      *e;
-  int len;
+  char_u *arg = skiptowhite(eap->arg);
+  int len     = (int)(arg - eap->arg);
+  arg         = skipwhite(arg);
 
-  e = skiptowhite(eap->arg);
-  len = (int)(e - eap->arg);
-  e = skipwhite(e);
+  const char *cmd = (char *)eap->arg;
 
-  if (len == 5 && STRNCMP(eap->arg, "start", 5) == 0 && *e != NUL) {
+  if (len == 5 && !strncmp(cmd, "start", 5) && *arg) {
```

Comment:
Use `NUL` when it is semantically appropriate. Also I believe most code (even new code) that I have seen compares the explicit result of `strcmp` family of functions, for clarity. 

As far as `STRNCMP` vs `strncmp`, we will make a mass-action decision about this later.